pkg/controllers/repositoryserver/handler.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jpillora/backoff"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kanisterio/kanister/pkg/kopia/command/storage"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/poll"
+)
+
+type RepoServerHandler struct {
+	Req                     ctrl.Request
+	Logger                  logr.Logger
+	Reconciler              *RepositoryServerReconciler
+	KubeCli                 kubernetes.Interface
+	RepositoryServer        *crv1alpha1.RepositoryServer
+	RepositoryServerSecrets repositoryServerSecrets
+}
+
+func (h *RepoServerHandler) CreateOrUpdateOwnedResources(ctx context.Context) error {
+	svc, err := h.reconcileService(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to reconcile service")
+	}
+	if err = h.getSecretsFromCR(ctx); err != nil {
+		return errors.Wrap(err, "failed to get repository server secrets")
+	}
+	pod, err := h.reconcilePod(ctx, svc)
+	if err != nil {
+		return errors.Wrap(err, "failed to reconcile repository server pod")
+	}
+	if err := h.waitForPodReady(ctx, pod); err != nil {
+		return errors.Wrap(err, "repository server pod not in ready state")
+	}
+	if err := h.connectToKopiaRepository(); err != nil {
+		return errors.Wrap(err, "failed to connect to kopia repository")
+	}
+
+	if err := h.startRepoProxyServer(ctx); err != nil {
+		return errors.Wrap(err, "failed to start kopia repository servver")
+	}
+
+	if err := h.createOrUpdateClientUsers(ctx); err != nil {
+		return errors.Wrap(err, "failed to create/update kopia server access users")
+	}
+	return nil
+}
+
+func (h *RepoServerHandler) reconcileService(ctx context.Context) (*corev1.Service, error) {
+	repoServerNamespace := h.RepositoryServer.Namespace
+	serviceName := h.RepositoryServer.Status.ServerInfo.ServiceName
+	svc := &corev1.Service{}
+	h.Logger.Info("Check if Service resource exists. If exists, reconcile with CR spec")
```

Comment:
```suggestion
	h.Logger.Info("Check if service resource exists. If exists, reconcile with CR spec")
```
[nit] There's inconsistency in the use of the resource types - I see both `Service` and `service` or `Pod`/`pod`. Let's again stick to one style.